### PR TITLE
adding gradle env variables to enable gradle entreprise

### PR DESCRIPTION
**Pull request by volatilemolotov**
Adds required variables for Gradle enterprise
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PreCommit_Go.yml [hunk 1]
```diff
@@ -49,6 +49,11 @@ permissions:
 concurrency:
   group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.sender.login }}-${{ github.event.schedule }}'
   cancel-in-progress: true
+env:
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
```
[the remark this answers is not in the document]
Comment (volatilemolotov) [reply .github/workflows/beam_PreCommit_Go.yml:53]: Should be added to all but by choice we do this one first to verify it works. 

After this is confirmed a new PR will be created for all the other jobs that will use caching

[the remark this answers is not in the document]
Comment (Abacn) [reply .github/workflows/beam_PreCommit_Go.yml:53]: I see, then probably its good to test with a Java precommit job. I am not sure if golang compile ever touches gradle cache

[the remark this answers is not in the document]
Comment (volatilemolotov) [reply .github/workflows/beam_PreCommit_Go.yml:53]: Im unfortunatley not aware of the scope the gradle cache is used, this was the job i have been using to test various issues. If we need another lets close this one and make another

**Comment by clayburn** [comment]
@volatilemolotov - I don't believe this will test the functionality unfortunately. GitHub Actions builds cannot access secrets from forks, so I don't think any of the authentication will work in this scenario.

**Comment by volatilemolotov** [comment]
> @volatilemolotov - I don't believe this will test the functionality unfortunately. GitHub Actions builds cannot access secrets from forks, so I don't think any of the authentication will work in this scenario.

The idea is to test it once it hits master in apache. 

Not sure of any way to test it properly until that? Not sure if i can push branches to apache/beam plus will the branch version be read for dispatch?
Also as @Abacn noted this job might not be the greatest for testing the gradle caching functionality

**Comment by clayburn** [comment]
Ah I see. Yes, if this is run from any branch within `apache/beam` then it should work.

**Comment by volatilemolotov** [comment]
@clayburn what i know is that its possible to run from branch from github cli, never tried it and im not sure if i should have the nedded permissions.

**Comment by Abacn** [comment]
superceded by #28212. Thanks @clayburn !

**Pull request closed without merging**